lib/gherkin_lint/linter/required_tags_starts_with.rb

```diff
@@ -0,0 +1,20 @@
+require 'gherkin_lint/linter/tag_constraint'
+require 'gherkin_lint/linter'
+
+module GherkinLint
+  # service class to lint for tags used multiple times
+  class RequiredTagsStartsWith < Linter
+    include TagConstraint
+
+    def match_pattern?(target)
+      match = false
+
+      target.each do |t|
+        t.delete! '@'
+        match = t.start_with?(*@pattern)
+        break if match
```

Comment:
would require a nicer error message like `Tag @foo does not match Pattern @MCC-[\d]+`